src/main/java/org/opensearch/neuralsearch/processor/AgenticQueryTranslatorProcessor.java

```diff
@@ -124,7 +131,28 @@ private void executeAgentAsync(
             try {
                 log.info("Generated Query: [{}]", agentResponse);
 
-                // TODO: Add query validations
+                // Validate response size to prevent memory exhaustion
+                if (agentResponse == null) {
+                    log.error(
+                        "Agentic search failed - Null response from agent - Agent ID: [{}], Query: [{}]",
+                        agentId,
+                        agenticQuery.getQueryText()
+                    );
+                    throw new IllegalArgumentException("Agent response is null.");
```

Comment:
If we are throwing exception already with a message then why we need log.error above?